src/backends/mod.rs

```diff
@@ -7,25 +7,50 @@ use crate::{
     compiler::Sources,
     constants::Span,
     error::{Error, ErrorKind, Result},
+    helpers::PrettyField,
     imports::FnHandle,
     var::{CellVar, Value, Var},
     witness::WitnessEnv,
 };
 
-use self::{kimchi::KimchiVesta, r1cs::R1csBls12_381};
+use self::{
+    kimchi::KimchiVesta,
+    r1cs::{R1csBls12381Field, R1csBn128Field, R1CS},
+};
 
 pub mod kimchi;
 pub mod r1cs;
 
+/// This trait serves as an alias for a bundle of traits
+pub trait BackendField:
+    Field + FromStr + TryFrom<BigUint> + TryInto<BigUint> + Into<BigUint> + PrettyField
+{
+}
+
 pub enum BackendKind {
     KimchiVesta(KimchiVesta),
-    R1CS(R1csBls12_381),
+    R1csBls12_381(R1CS<R1csBls12381Field>),
+    R1csBn128(R1CS<R1csBn128Field>),
```

Comment:
should we call it bn254 everywhere? what's the most popular name :D?

Reply:
snarkjs calls it bn128 :)

Reply:
maybe better with bn254 as the google results seem to prefer this one